commons/src/main/java/org/dhis2/commons/resources/ResourceManager.kt

```diff
@@ -109,24 +109,58 @@ class ResourceManager(
             .getErrorMessage(throwable)
 
     fun defaultEventLabel(): String = getWrapperContext().getString(R.string.events)
-    fun getEventLabel(
+    private fun getProgramStageEventLabel(
         programStageUid: String? = null,
+        programUid: String? = null,
         quantity: Int = 1,
     ) = try {
         D2Manager.getD2().programModule()
             .programStages().uid(programStageUid)
-            .blockingGet()?.eventLabel()
+            .blockingGet()?.displayEventLabel()
     } catch (e: Exception) {
         null
-    } ?: getPlural(R.plurals.event_label, quantity)
+    } ?: programEventLabel(programUid, quantity)
 
-    fun formatWithEventLabel(
+    fun formatWithProgramStageEventLabel(
```

Comment:
If we add theses specific labels in the resource manager it will grow too much.
What about create a `EventResourcersProvider` to provide all resources of the event, and in the future the same for program etc... with this we develop the code once per label injecting the `resourcesManager` to get the strings, with this we leave resourcesManager just to get resources and we develop one provider per scope. What do you think?

Reply:
Yes, definitely a better approach. I will try to refactor this, thanks for the idea!